src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -383,12 +383,19 @@ def _get_dependency_distance(var_name, index_read, index_written):
             return None
 
         symbol_map = sympy_writer.type_map
-        # If the subscripts do not even depend on the specified variable,
-        # any dependency distance is possible (e.g. `do i ... a(j)=a(j)+1`)
-        if var_name not in symbol_map:
+        # Find the SymPy symbol that has the same name as the var name. We
+        # cannot use the dictionary key, since a symbol might be renamed
+        # (e.g. if a Fortran variable 'lambda' is used, it will be
+        # renamed to lambda_1, and the type map will have lambda_1 as key
+        # and the SymPy symbol for 'lambda' as value).
+        for var in symbol_map.values():
```

Comment:
Can't you just try to access it in a try with a KeyError?
```
try:
    result = symbol_map[var_name]
except KeyError:
    ...
```

Reply:
No, I hoped that my explanation was good enough: e.g. in the case of `lambda`, the key will be `lambda_1`, but the value `lambda`.  So I would need to use a reverse dictionary to look it up, but I am not sure if this is worth computing.